packages/syft/src/syft/service/enclave/domain_enclave_service.py

```diff
@@ -0,0 +1,220 @@
+# stdlib
+import itertools
+from typing import Any
+
+# relative
+from ...serde.serializable import serializable
+from ...service.action.action_permissions import ActionObjectPermission
+from ...service.action.action_permissions import ActionPermission
+from ...service.network.routes import NodeRouteType
+from ...service.response import SyftError
+from ...service.response import SyftSuccess
+from ...service.user.user_roles import ADMIN_ROLE_LEVEL
+from ...service.user.user_roles import DATA_SCIENTIST_ROLE_LEVEL
+from ...store.document_store import DocumentStore
+from ...types.uid import UID
+from ..code.user_code import UserCode
+from ..context import AuthedServiceContext
+from ..service import AbstractService
+from ..service import service_method
+from .enclave import EnclaveInstance
+from .enclave_stash import EnclaveInstanceStash
+
+
+@serializable()
+class DomainEnclaveService(AbstractService):
+    """Contains service methods for Domain -> Enclave communication."""
+
+    store: DocumentStore
+    stash: EnclaveInstanceStash
+
+    def __init__(self, store: DocumentStore) -> None:
+        self.store = store
+        self.stash = EnclaveInstanceStash(store=store)
+
+    @service_method(
+        path="enclave.add",
```

Comment:
should these service methods maintain the same pathing as previously or be something like `domain_enclave.add` now so it's clear which service is being called?

Reply:
That is a valid point, I am also  not fully sure, for now, we could leave and we could think about it, when we shift to Old Project class.

Reply:
+1 to this. I am not able to come up with a decent name though. Any suggestions will be appreciated. Otherwise we can park this one for later.

Reply:
Changing to `enclave_internal.*` for internal paths and `enclave.*` for open paths?

Reply:
I like `enclave_internal.*`. It might not work if we decide to give direct Enclave access to the DS though.